From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 27 Jan 2017 04:53:54 +0000
Subject: arm64: ptrace: Avoid ABI change in 4.9.6
Forwarded: not-needed

Commit aeb1f39d814b "arm64/ptrace: Avoid uninitialised struct padding
in fpr_set()" added a new member to struct user_fpsimd_state, but it
takes the place of what was padding (for 128-bit alignment).  Hide
this from genksyms.

---
--- a/arch/arm64/include/uapi/asm/ptrace.h
+++ b/arch/arm64/include/uapi/asm/ptrace.h
@@ -77,7 +77,9 @@ struct user_fpsimd_state {
 	__uint128_t	vregs[32];
 	__u32		fpsr;
 	__u32		fpcr;
+#if !defined(__KERNEL__) || !defined(__GENKSYMS__)
 	__u32		__reserved[2];
+#endif
 };
 
 struct user_hwdebug_state {
